src/validation.cpp

```diff
@@ -2206,13 +2206,30 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
     return true;
 }
 
+CoinsCacheSizeState CChainState::GetCoinsCacheSizeState()
+{
+    static int64_t nMempoolSizeMax = gArgs.GetArg("-maxmempool", DEFAULT_MAX_MEMPOOL_SIZE) * 1000000;
```

Comment:
Maybe avoid introducing `static` here since there was no `static` previously, and in general this isn't a pattern we use with other `GetArg` calls, and and it makes this code harder to test with different sizes, or allow updating the size without restarting.

Reply:
Good points, will fix. Can't remember why I added that.